pkg/updates/checker.go

```diff
@@ -76,6 +76,40 @@ const (
 	updateInterval       = 30 * time.Minute
 )
 
+// TryGetAnonymousID returns the instance ID from the updates file if it exists.
+// This is a read-only operation - it never generates a new ID.
+// Returns empty string if the file doesn't exist or doesn't contain an instance ID.
+// Use this for optional features like metrics that shouldn't trigger ID generation.
+// TODO this should probably be extracted into its own package to handle instance ID generation.
+func TryGetAnonymousID() (string, error) {
```

Comment:
Can we cache this ID in memory for the duration of the process? this way, we'd avoid reading the file every 15 mins.